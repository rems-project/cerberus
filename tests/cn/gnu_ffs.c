int ffs(int x)
/*@ ensures (x == 0i32) ? (return == 0i32) : true;
            (x == 1i32) ? (return == 1i32) : true;
            (x == 2i32) ? (return == 2i32) : true;
            (x == 3i32) ? (return == 1i32) : true;
            (x == 8i32) ? (return == 4i32) : true; @*/
{
  return __builtin_ffs(x);
}

int main(void) {
  int r = ffs(1);
  return 0;
}