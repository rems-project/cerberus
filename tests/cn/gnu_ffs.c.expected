return code: 0
[1/2]: ffs -- pass
[2/2]: main -- pass
